### DISCUSSION TITLE
blog/2022/11/29/conditional-marginal-marginaleffects/index

### DISCUSSION THREAD

**Discussion by giscus** [andrewheiss/ath-quarto#76] [Blog comments]
# blog/2022/11/29/conditional-marginal-marginaleffects/index

Use the {marginaleffects} package to calculate tricky and nuanced marginal and conditional effects in generalized linear mixed models

https://www.andrewheiss.com/blog/2022/11/29/conditional-marginal-marginaleffects/index.html

**Comment by sstew45** [comment]
The detail used here in teasing out the different estimates, and understanding their interpretation, is really nice to see. What's even better is how you summarized each part of the analysis in a beautiful and digestable narrative, complete with journal-quality tables and plots and well-organized code. I will be bookmarking this for multiple use-cases. Great post!

**Comment by epkanol** [comment]
Sorry if I'm dumb, but I'm trying to get my head around Bayesian inference, and your excellent posts help me a lot in that regard, Andrew.

When you say "There are a few problems with these estimates though: (1) they’re on the log odds scale, "
shouldn't it really be "...on the *log* scale..."

(No less hard to interpret, it's just that I reserved "log odds" for the "logit" link in my brain, and here we are using a log link, i.e. the lognormal distribution)

**Comment by nahorp** [comment]
Thanks once again Andrew! What would the `emmeans` equivalent of the `marginaleffects` code look like? Even if only in the summary table, it'd be great to see a comparison!

**Comment by gugushvili** [comment]
Thanks for a nice blog post. Just wanted to point out [this article]( https://doi.org/10.1111/j.1467-842X.2004.00334.x) (page 327), where the terms conditional and marginal are defined in the opposite way to what you wrote: there marginal predictions are obtained by setting random offsets to zero. They are not the only ones to do it.

**Comment by DanSimonet** [comment]
Hey Andrew awesome blog posts - your marginalia overview is one of my favorites! Question for your code  on `Population-level ATE (marginalized)` . Should it also include the unique clusters in data grid like your example for `Average population-level outcomes (marginalized)`? So like this

```
comparisons(
  fit, 
  variables = "TX", 
  newdata = datagrid(TX = c(0, 1), 
                     cluster = unique), 
  re_formula = NULL
) %>% 
  tidy()
```
I receive very different results for some test cases if the `cluster=unique` is omitted. Thanks!
